gdal/alg/gdalwarpkernel.cpp

```diff
@@ -6156,10 +6154,8 @@ static void GWKAverageOrModeThread( void* pData)
                     {
                         if (bIsComplex)
                         {
-                            dfR = dfTotalReal / dfTotalWeight;
-                            dfI = dfTotalImag / dfTotalWeight;
-                            dfValueReal = sqrt((dfR + sqrt(dfR * dfR + dfI * dfI))/2.);;
-                            dfValueImag = dfI / fabs(dfI) * sqrt((-dfR + sqrt(dfR * dfR + dfI * dfI)) / 2.);
+                            dfValueReal = sqrt( dfTotalReal / dfTotalWeight );
```

Comment:
the computation of dfTotalReal and dfTotalImag done a bit above in the bIsComplex case should be fixed to be similar to the one in the !bIsComplex case. Same remark for overview.cpp

Reply:
done